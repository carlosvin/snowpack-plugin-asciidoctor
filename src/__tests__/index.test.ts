import path from 'path'
import { promises as fs } from 'fs'
const snowpackPluginAdoc = require('../index.ts')
import MockDate from 'mockdate'

MockDate.set(1482363367071)

describe('snowpack-plugin-asciidoctor', () => {
  it('should compile .adoc files', async () => {
    const filePath = path.join(__dirname, '__fixtures__/test.adoc')
    const plugin = snowpackPluginAdoc({}, {})
    const result = await plugin.load({ filePath })
    expect(result).toMatchSnapshot()
  })

  it('should add asciidoctor attributes', async () => {
    const filePath = path.join(__dirname, '__fixtures__/a-post.adoc')
    const plugin = snowpackPluginAdoc(
      {},
      {
        authorinitials: 'CMS',
        copyright: 'Carlos MS 2020',
        toclevels: 5,
      },
    )
    const result = await plugin.load({ filePath })
    expect(result).toMatchSnapshot()
  })
})
