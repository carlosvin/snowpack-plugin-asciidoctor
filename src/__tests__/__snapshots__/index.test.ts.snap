// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snowpack-plugin-asciidoctor should add asciidoctor attributes 1`] = `
Object {
  ".js": Object {
    "code": "export const doc = {
            filePath: \\"/home/carlos/workspace/snowpack-plugin-asciidoctor/src/__tests__/__fixtures__/a-post.adoc\\",
            html: \`<div id=\\"toc\\" class=\\"toc\\">
<div id=\\"toctitle\\">Table of Contents</div>
<ul class=\\"sectlevel1\\">
<li><a href=\\"#_sapper_1_0_is_not_going_to_happen\\">Sapper 1.0 is not going to happen</a></li>
<li><a href=\\"#_snowpack\\">Snowpack</a></li>
<li><a href=\\"#_example\\">Example</a>
<ul class=\\"sectlevel2\\">
<li><a href=\\"#_build\\">Build</a>
<ul class=\\"sectlevel3\\">
<li><a href=\\"#_ssr\\">SSR</a></li>
<li><a href=\\"#_pure_static_site\\">Pure static site</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div id=\\"preamble\\">
<div class=\\"sectionbody\\">
<div class=\\"paragraph\\">
<p>Today I finally got some time to watch the video <a href=\\"https://www.youtube.com/watch?v=qSfdtmcZ4d0\\" target=\\"_blank\\" rel=\\"noopener\\">Rich Harris: Futuristic Web Development</a> recorded at <a href=\\"https://sveltesummit.com/\\" target=\\"_blank\\" rel=\\"noopener\\">Svelte Summit 2020</a>. If you have 20 minutes, just stop reading and <a href=\\"https://www.youtube.com/watch?v=qSfdtmcZ4d0\\" target=\\"_blank\\" rel=\\"noopener\\">watch it</a>.</p>
</div>
<div class=\\"admonitionblock tip\\">
<table>
<tr>
<td class=\\"icon\\">
<div class=\\"title\\">Tip</div>
</td>
<td class=\\"content\\">
If you have 20 minutes, just stop reading and watch <a href=\\"https://www.youtube.com/watch?v=qSfdtmcZ4d0\\" target=\\"_blank\\" rel=\\"noopener\\">Rich Harris: Futuristic Web Development</a> video.
</td>
</tr>
</table>
</div>
</div>
</div>
<div class=\\"sect1\\">
<h2 id=\\"_sapper_1_0_is_not_going_to_happen\\">Sapper 1.0 is not going to happen</h2>
<div class=\\"sectionbody\\">
<div class=\\"paragraph\\">
<p>As you might know, this blog is powered by <a href=\\"https://sapper.svelte.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Sapper</a>. I am already quite happy with it, so when I hear this announcement, I felt like when they cancel a TV Series that I am enjoying.</p>
</div>
<div class=\\"paragraph\\">
<p>Happily, there is a good reason, there is going to be a better <a href=\\"https://sapper.svelte.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Sapper</a>, I think they will call it svelte-kit, it is solving some issues and improving some aspects of <a href=\\"https://sapper.svelte.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Sapper</a>, but the main benefit, in my opinion, is that you won&#8217;t have to choose between <a href=\\"https://sapper.svelte.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Sapper</a> or <a href=\\"https://svelte.dev/\\" target=\\"_blank\\" rel=\\"noopener\\">Svelte</a> when you start a new application, everything will be supported by the <a href=\\"https://svelte.dev/\\" target=\\"_blank\\" rel=\\"noopener\\">Svelte</a> ecosystem.</p>
</div>
</div>
</div>
<div class=\\"sect1\\">
<h2 id=\\"_snowpack\\">Snowpack</h2>
<div class=\\"sectionbody\\">
<div class=\\"paragraph\\">
<p><a href=\\"https://www.snowpack.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Snowpack</a> will become the default <a href=\\"https://svelte.dev/\\" target=\\"_blank\\" rel=\\"noopener\\">Svelte</a> builder. It is not a regular bundler like <a href=\\"https://webpack.js.org\\" target=\\"_blank\\" rel=\\"noopener\\">Webpack</a> or <a href=\\"https://rollupjs.org/\\" target=\\"_blank\\" rel=\\"noopener\\">Rollup</a>. It relies on <a href=\\"https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import\\" target=\\"_blank\\" rel=\\"noopener\\">Javascript modules</a> so your application delegate the modules loading on the web browser; traditionally the chunks are loaded by the bundler (or by source code injected by the bundler).</p>
</div>
</div>
</div>
<div class=\\"sect1\\">
<h2 id=\\"_example\\">Example</h2>
<div class=\\"sectionbody\\">
<div class=\\"admonitionblock note\\">
<table>
<tr>
<td class=\\"icon\\">
<div class=\\"title\\">Note</div>
</td>
<td class=\\"content\\">
This example is just a copy from the <a href=\\"https://www.youtube.com/watch?v=qSfdtmcZ4d0\\" target=\\"_blank\\" rel=\\"noopener\\">Rich Harris: Futuristic Web Development</a> video.
</td>
</tr>
</table>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">How to create the sample project</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">mkdir svelte-next-sample
<span class=\\"hljs-built_in\\">cd</span> svelte-next-sample
npm init svelte@next
npm install</code></pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">Start development server</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">npm run dev

[snowpack] &gt; Listening on http://localhost:3000</code></pre>
</div>
</div>
<div class=\\"paragraph\\">
<p>You will find many similarities with <a href=\\"https://sapper.svelte.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Sapper</a>, like the <code>routes</code> and <code>components</code> folders.</p>
</div>
<div class=\\"sect2\\">
<h3 id=\\"_build\\">Build</h3>
<div class=\\"listingblock\\">
<div class=\\"title\\">Build the project for production</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">npm run build</code></pre>
</div>
</div>
<div class=\\"sect3\\">
<h4 id=\\"_ssr\\">SSR</h4>
<div class=\\"paragraph\\">
<p>By default the project is configured with SSR rendering enabled.</p>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">SSR with nodejs server and client</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">npm run build
...

&gt; Optimizing...
  ✔ server
  ✔ client

&gt; Generating app...
  Using @sveltejs/adapter-node <b class=\\"conum\\">(1)</b>
  Prerendering static pages...
  ✔ <span class=\\"hljs-keyword\\">done</span></code></pre>
</div>
</div>
<div class=\\"colist arabic\\">
<ol>
<li>
<p>This is the default adapter to generate SSR application with a nodejs server and a client.</p>
</li>
</ol>
</div>
</div>
<div class=\\"sect3\\">
<h4 id=\\"_pure_static_site\\">Pure static site</h4>
<div class=\\"paragraph\\">
<p>This blog is a pure static website, it doesn&#8217;t require a server. With <a href=\\"https://sapper.svelte.dev\\" target=\\"_blank\\" rel=\\"noopener\\">Sapper</a> we have the option to execute <code>sapper export</code> and it generates the app with no server required.</p>
</div>
<div class=\\"paragraph\\">
<p>With this new approach, to generate a static site, we just have to use a different adapter:</p>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">npm i -D @sveltejs/adapter-static</code></pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">svelte.config.js</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-javascript\\" data-lang=\\"javascript\\"><span class=\\"hljs-built_in\\">module</span>.exports = {
	<span class=\\"hljs-attr\\">adapter</span>: <span class=\\"hljs-string\\">'@sveltejs/adapter-static'</span>
};</code></pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">npm run build

&gt; Generating app...
  Using @sveltejs/adapter-static <b class=\\"conum\\">(1)</b>
  ✔ <span class=\\"hljs-keyword\\">done</span></code></pre>
</div>
</div>
<div class=\\"colist arabic\\">
<ol>
<li>
<p>The static adapter is selected</p>
</li>
</ol>
</div>
</div>
</div>
</div>
</div>\`,
            metadata: {
  \\"attribute-undefined\\": \\"drop-line\\",
  \\"attribute-missing\\": \\"skip\\",
  \\"appendix-caption\\": \\"Appendix\\",
  \\"appendix-refsig\\": \\"Appendix\\",
  \\"caution-caption\\": \\"Caution\\",
  \\"chapter-refsig\\": \\"Chapter\\",
  \\"example-caption\\": \\"Example\\",
  \\"figure-caption\\": \\"Figure\\",
  \\"important-caption\\": \\"Important\\",
  \\"last-update-label\\": \\"Last updated\\",
  \\"note-caption\\": \\"Note\\",
  \\"part-refsig\\": \\"Part\\",
  \\"prewrap\\": \\"\\",
  \\"sectids\\": \\"\\",
  \\"section-refsig\\": \\"Section\\",
  \\"table-caption\\": \\"Table\\",
  \\"tip-caption\\": \\"Tip\\",
  \\"toc-placement\\": \\"auto\\",
  \\"toc-title\\": \\"Table of Contents\\",
  \\"untitled-label\\": \\"Untitled\\",
  \\"version-label\\": \\"Version\\",
  \\"warning-caption\\": \\"Warning\\",
  \\"notitle\\": \\"\\",
  \\"source-highlighter\\": \\"highlightjs-ext\\",
  \\"authorinitials\\": \\"CMS\\",
  \\"copyright\\": \\"Carlos MS 2020\\",
  \\"toclevels\\": 5,
  \\"docfile\\": \\"/home/carlos/workspace/snowpack-plugin-asciidoctor/src/__tests__/__fixtures__/a-post.adoc\\",
  \\"docdir\\": \\"/home/carlos/workspace/snowpack-plugin-asciidoctor/src/__tests__/__fixtures__\\",
  \\"docfilesuffix\\": \\".adoc\\",
  \\"docname\\": \\"a-post\\",
  \\"embedded\\": \\"\\",
  \\"asciidoctor\\": \\"\\",
  \\"asciidoctor-version\\": \\"2.0.12\\",
  \\"safe-mode-name\\": \\"unsafe\\",
  \\"safe-mode-unsafe\\": \\"\\",
  \\"safe-mode-level\\": 0,
  \\"max-include-depth\\": 64,
  \\"user-home\\": \\"/home/carlos\\",
  \\"doctype\\": \\"article\\",
  \\"htmlsyntax\\": \\"html\\",
  \\"backend-html5-doctype-article\\": \\"\\",
  \\"doctype-article\\": \\"\\",
  \\"backend-html5\\": \\"\\",
  \\"backend\\": \\"html5\\",
  \\"outfilesuffix\\": \\".html\\",
  \\"filetype\\": \\"html\\",
  \\"filetype-html\\": \\"\\",
  \\"basebackend-html-doctype-article\\": \\"\\",
  \\"basebackend-html\\": \\"\\",
  \\"basebackend\\": \\"html\\",
  \\"stylesdir\\": \\".\\",
  \\"iconsdir\\": \\"./images/icons\\",
  \\"localdate\\": \\"2016-12-22\\",
  \\"localyear\\": \\"2016\\",
  \\"localtime\\": \\"00:36:07 +0100\\",
  \\"localdatetime\\": \\"2016-12-22 00:36:07 +0100\\",
  \\"docdate\\": \\"2020-12-06\\",
  \\"docyear\\": \\"2020\\",
  \\"doctime\\": \\"20:16:33 +0100\\",
  \\"docdatetime\\": \\"2020-12-06 20:16:33 +0100\\",
  \\"doctitle\\": \\"Svelte, a bright future with Snowpack\\",
  \\"date\\": \\"2020-10-25\\",
  \\"keywords\\": \\"Svelte, Sapper, Snowpack\\",
  \\"lang\\": \\"en\\",
  \\"description\\": \\"Echoing Richard Harris announcements in Svelte Summit 2020: Sapper 1.0 never gonna happen, the new Svelte Kit, Snowpack as build tool, Javascript Module System.\\",
  \\"toc\\": \\"\\",
  \\"link\\": \\"https://www.youtube.com/watch?v=qSfdtmcZ4d0\\",
  \\"authorcount\\": 0,
  \\"svelte-summit\\": \\"https://sveltesummit.com/[Svelte Summit 2020,window=_blank]\\",
  \\"sapper\\": \\"https://sapper.svelte.dev[Sapper,window=_blank]\\",
  \\"svelte\\": \\"https://svelte.dev/[Svelte,window=_blank]\\",
  \\"snowpack\\": \\"https://www.snowpack.dev[Snowpack,window=_blank]\\",
  \\"js-modules\\": \\"https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import[Javascript modules,window=_blank]\\",
  \\"webpack\\": \\"https://webpack.js.org[Webpack,window=_blank]\\",
  \\"rollup\\": \\"https://rollupjs.org/[Rollup,window=_blank]\\",
  \\"video\\": \\"https://www.youtube.com/watch?v=qSfdtmcZ4d0[Rich Harris: Futuristic Web Development,window=_blank]\\"
}
          };",
  },
}
`;

exports[`snowpack-plugin-asciidoctor should compile .adoc files 1`] = `
Object {
  ".js": Object {
    "code": "export const doc = {
            filePath: \\"/home/carlos/workspace/snowpack-plugin-asciidoctor/src/__tests__/__fixtures__/test.adoc\\",
            html: \`<div class=\\"paragraph\\">
<p>It’s fair to say that React 16.8 and the introduction of
<a href=\\"https://reactjs.org/docs/hooks-intro.html\\">hooks</a> has really changed how
we write React. Hooks are one of those APIs that make you realise the
flaws of the previous approach <em>after</em> you stop using it. I remember
being very skeptical of hooks when they were first released, not
thinking that the previous class based design had many flaws, but I’ve
since come to realise I was very wrong, and hooks are a vast improvement
on how we build React components. If you’re interested in comparing the
old vs the new, I wrote a <a href=\\"/refactoring-to-react-hooks/\\">blog post
refactoring a component to use hooks</a> that offers a nice comparison.</p>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-py\\" data-lang=\\"py\\"><span class=\\"hljs-function\\"><span class=\\"hljs-keyword\\">def</span> <span class=\\"hljs-title\\">main</span><span class=\\"hljs-params\\">()</span>:</span>
    <span class=\\"hljs-keyword\\">pass</span></code></pre>
</div>
</div>
<div class=\\"admonitionblock warning\\">
<table>
<tr>
<td class=\\"icon\\">
<div class=\\"title\\">Warning</div>
</td>
<td class=\\"content\\">
Wolpertingers are known to nest in server racks.
Enter at your own risk.
</td>
</tr>
</table>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">void main</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-cpp\\" data-lang=\\"cpp\\"><span class=\\"hljs-function\\"><span class=\\"hljs-keyword\\">int</span> <span class=\\"hljs-title\\">main</span> <span class=\\"hljs-params\\">()</span> </span>{
    <span class=\\"hljs-keyword\\">return</span> <span class=\\"hljs-number\\">0</span>;
}</code></pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">bash file include</div>
<div class=\\"content\\">
<pre class=\\"highlight\\"><code class=\\"language-bash\\" data-lang=\\"bash\\">FILES=*.md
<span class=\\"hljs-keyword\\">for</span> f <span class=\\"hljs-keyword\\">in</span> <span class=\\"hljs-variable\\">$FILES</span>
<span class=\\"hljs-keyword\\">do</span>
  filename=<span class=\\"hljs-string\\">\\"<span class=\\"hljs-variable\\">\${f%.*}</span>\\"</span>
  <span class=\\"hljs-built_in\\">echo</span> <span class=\\"hljs-string\\">\\"Converting <span class=\\"hljs-variable\\">$f</span> to <span class=\\"hljs-variable\\">$filename</span>.adoc\\"</span>
  \`pandoc <span class=\\"hljs-variable\\">$f</span> -f markdown -t asciidoctor -o <span class=\\"hljs-variable\\">$filename</span>.adoc\`
<span class=\\"hljs-keyword\\">done</span></code></pre>
</div>
</div>\`,
            metadata: {
  \\"attribute-undefined\\": \\"drop-line\\",
  \\"attribute-missing\\": \\"skip\\",
  \\"appendix-caption\\": \\"Appendix\\",
  \\"appendix-refsig\\": \\"Appendix\\",
  \\"caution-caption\\": \\"Caution\\",
  \\"chapter-refsig\\": \\"Chapter\\",
  \\"example-caption\\": \\"Example\\",
  \\"figure-caption\\": \\"Figure\\",
  \\"important-caption\\": \\"Important\\",
  \\"last-update-label\\": \\"Last updated\\",
  \\"note-caption\\": \\"Note\\",
  \\"part-refsig\\": \\"Part\\",
  \\"prewrap\\": \\"\\",
  \\"sectids\\": \\"\\",
  \\"section-refsig\\": \\"Section\\",
  \\"table-caption\\": \\"Table\\",
  \\"tip-caption\\": \\"Tip\\",
  \\"toc-placement\\": \\"auto\\",
  \\"toc-title\\": \\"Table of Contents\\",
  \\"untitled-label\\": \\"Untitled\\",
  \\"version-label\\": \\"Version\\",
  \\"warning-caption\\": \\"Warning\\",
  \\"notitle\\": \\"\\",
  \\"source-highlighter\\": \\"highlightjs-ext\\",
  \\"docfile\\": \\"/home/carlos/workspace/snowpack-plugin-asciidoctor/src/__tests__/__fixtures__/test.adoc\\",
  \\"docdir\\": \\"/home/carlos/workspace/snowpack-plugin-asciidoctor/src/__tests__/__fixtures__\\",
  \\"docfilesuffix\\": \\".adoc\\",
  \\"docname\\": \\"test\\",
  \\"embedded\\": \\"\\",
  \\"asciidoctor\\": \\"\\",
  \\"asciidoctor-version\\": \\"2.0.12\\",
  \\"safe-mode-name\\": \\"unsafe\\",
  \\"safe-mode-unsafe\\": \\"\\",
  \\"safe-mode-level\\": 0,
  \\"max-include-depth\\": 64,
  \\"user-home\\": \\"/home/carlos\\",
  \\"doctype\\": \\"article\\",
  \\"htmlsyntax\\": \\"html\\",
  \\"backend-html5-doctype-article\\": \\"\\",
  \\"doctype-article\\": \\"\\",
  \\"backend-html5\\": \\"\\",
  \\"backend\\": \\"html5\\",
  \\"outfilesuffix\\": \\".html\\",
  \\"filetype\\": \\"html\\",
  \\"filetype-html\\": \\"\\",
  \\"basebackend-html-doctype-article\\": \\"\\",
  \\"basebackend-html\\": \\"\\",
  \\"basebackend\\": \\"html\\",
  \\"stylesdir\\": \\".\\",
  \\"iconsdir\\": \\"./images/icons\\",
  \\"localdate\\": \\"2016-12-22\\",
  \\"localyear\\": \\"2016\\",
  \\"localtime\\": \\"00:36:07 +0100\\",
  \\"localdatetime\\": \\"2016-12-22 00:36:07 +0100\\",
  \\"docdate\\": \\"2020-11-22\\",
  \\"docyear\\": \\"2020\\",
  \\"doctime\\": \\"21:22:18 +0100\\",
  \\"docdatetime\\": \\"2020-11-22 21:22:18 +0100\\",
  \\"doctitle\\": \\"My Document's Title\\",
  \\"firstname\\": \\"Doc\\",
  \\"authorinitials\\": \\"DW\\",
  \\"lastname\\": \\"Writer\\",
  \\"author\\": \\"Doc Writer\\",
  \\"email\\": \\"doc.writer@asciidoctor.org\\",
  \\"authors\\": \\"Doc Writer\\",
  \\"authorcount\\": 1,
  \\"revnumber\\": \\"1.0\\",
  \\"revdate\\": \\"2014-01-01\\",
  \\"layout\\": \\"post\\",
  \\"intro\\": \\"A short post today about an easy tactic to avoid your useEffect calls becoming recursive when setting state.\\",
  \\"asdf\\": \\"made up attr\\"
}
          };",
  },
}
`;
